hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/offset/OffsetCommitter.java

```diff
@@ -107,21 +105,25 @@ public void run() {
             int scheduledToCommit = 0;
             OffsetsToCommit offsetsToCommit = new OffsetsToCommit();
             for (SubscriptionPartition partition : Sets.union(minInflightOffsets.keySet(), maxCommittedOffsets.keySet())) {
-                long offset = Math.min(
-                        minInflightOffsets.getOrDefault(partition, Long.MAX_VALUE),
-                        maxCommittedOffsets.getOrDefault(partition, Long.MAX_VALUE)
-                );
-                if (offset >= 0 && offset < Long.MAX_VALUE) {
-                    scheduledToCommit++;
-                    offsetsToCommit.add(new SubscriptionPartitionOffset(partition, offset));
+                if (partitionAssignmentState.isAssignedPartitionAtCurrentTerm(partition)) {
+                    long offset = Math.min(
+                            minInflightOffsets.getOrDefault(partition, Long.MAX_VALUE),
+                            maxCommittedOffsets.getOrDefault(partition, Long.MAX_VALUE)
```

Comment:
Should we've `Long.MIN_VALUE` here?

Reply:
No, we're looking for smallest value here, so if the default for missing value is MIN_VALUE, it always wins. :)